coderd/notifications_test.go

```diff
@@ -0,0 +1,79 @@
+package coderd_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestUpdateNotificationsSettings(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Permissions denied", func(t *testing.T) {
+		t.Parallel()
+
+		api := coderdtest.New(t, nil)
+		firstUser := coderdtest.CreateFirstUser(t, api)
+		anotherClient, _ := coderdtest.CreateAnotherUser(t, api, firstUser.OrganizationID)
+
+		// given
+		expected := codersdk.NotificationsSettings{
+			NotifierPaused: true,
+		}
+
+		ctx := testutil.Context(t, testutil.WaitShort)
+
+		// when
+		err := anotherClient.PutNotificationsSettings(ctx, expected)
+
+		// then
+		require.Error(t, err) // Insufficient permissions to update notifications settings.
```

Comment:
Let's check for the specific error we're expecting to receive to prevent false-positives.

Reply:
Fixed